tools/update_centreon_storage_logs.php

```diff
@@ -0,0 +1,391 @@
+                date('Y-m-d H:i:s'),
+                $partitionName,
+                $currentStep,
+                $message
+            );
+        } else {
+            printf(
+                "[%s] [STEP %d] %s\n",
+                date('Y-m-d H:i:s'),
+                $currentStep,
+                $message
+            );
+        }
+    } else {
+        printf("[%s] %s\n", date('Y-m-d H:i:s'), $message);
+    }
+};
+
+/**
+ * Show message but without showing the partition name and step
+ *
+ * @param string $message Message to show
+ */
+function mySysLog(string $message): void
+{
+    global $logs;
+    $logs($message, false);
+}
+
+/**
+ * Stop the Centreon Broker process
+ *
+ * @return int Return status of the Broker stop
+ */
+function stopBroker(): int
+{
+    exec('systemctl stop cbd', $output, $status);
+    return empty($output) ? $status : -1;
+}
+
+/**
+ * Start the Centreon Broker process
+ *
+ * @return int Return status of the Broker start
+ */
+function startBroker(): int
+{
+    exec('systemctl start cbd', $output, $status);
+
+    return empty($output) ? $status : -1;
+}
+
+/**
+ * Retrieve the started process number of Broker
+ *
+ * @return int Return the started process number
+ */
+function getBrokerProcess()
+{
+    exec(
+        'ps -o args -p $(pidof -o $$ -o $PPID -o %PPID -x cbd || echo 1000000) | grep -c /usr/sbin/cbd',
+        $result
+    );
+    return (int) $result[0];
+}
+
+/**
+ * Get all partitions that are not empty
+ *
+ * @param \PDO $db
+ * @return array Return an array of type [partition_name => numberOfRecords,...]
+ */
+function getNotEmptyPartitions(\PDO $db)
+{
+    $result = $db->query(
+        "SELECT PARTITION_NAME, TABLE_ROWS FROM INFORMATION_SCHEMA.PARTITIONS "
+        . "WHERE TABLE_NAME='logs'"
+    );
+    $partitions = [];
+    while (($row = $result->fetch(\PDO::FETCH_ASSOC))) {
+        $nbrRows = (int) $row['TABLE_ROWS'];
+        if ($nbrRows > 0) {
+            $partitions[$row['PARTITION_NAME']] = $row['TABLE_ROWS'];
+        }
+    }
+    ksort($partitions);
+    return $partitions;
+}
+
+$dbUser = 'root';
+
+$insertQuery = <<<'QUERY'
+LOAD DATA INFILE '{{DATA_FILE}}'
+INTO TABLE logs 
+FIELDS TERMINATED BY ','
+LINES TERMINATED BY '\n'
+(ctime, host_id, host_name, instance_name, issue_id, msg_type, notification_cmd, 
+notification_contact, output, retry, service_description, service_id, status, type)
+QUERY;
+
+// We create a loc file to avoid to start this script one more time
+$fileInfos = pathinfo(__FILE__);
+$lockFileName = $fileInfos['filename'] . '.loc';
+if (file_exists($lockFileName)) {
+    mySysLog('Process ' . __FILE__ . ' already running');
+    exit();
+}
+touch($lockFileName);
+
+$currentStep = 1;
+try {
+    // We check if the database connection port is set, otherwise we request it.
+    if (defined('port')) {
+        $dbPort = port;
+    } else {
+        $port = askQuestion("Please enter the port of the Centreon database : ");
```

Comment:
In modern English-language printing, no space is placed before a colon
https://en.wikipedia.org/wiki/Colon_(punctuation)#Spacing

Reply:
Fixed, I had forgotten this one.